api.md

```diff
@@ -170,6 +171,114 @@ SELECT add_dimension('conditions', 'device_id', number_partitions => 2);
 SELECT add_dimension('conditions', 'device_id', number_partitions => 2, if_not_exists => true);
 ```
 
+---
+## add_data_node() :community_function: [](add_data_node)
+
+WHen executed on the access node, this will add a new data node to the
+current database for use with distributed hypertables.
+
+Newly created distributed hypertables will be able to make use of this
+data node for distributing data across the cluster. Note that existing
+distributed hypertables will not use the newly added data node. For
+existing distributed hypertables to use added data nodes, you need to
+use [`attach_data_node()`](#attach_data_node).
+
+The function will attempt to bootstrap the data node by creating the
+database where the remote hypertables will later be created and also
+load the TimescaleDB extension on the data node.
+
+#### Errors
+
+* An error will be given if an attempt is made to run the function
+  inside a transaction.
+* An error will be given if the remote database already exists on the
+  data node and `if_not_exists` is not set.
+
+#### Required Arguments [](add_data_node-required-arguments)
+
+| Name        | Description            |
+| ----------- | -----------            |
+| `node_name` | Name of the data node. |
+
+##### Parameter Description
+
+* `node_name` - Name to use for the node locally.
+
+#### Optional Arguments [](add_data_node-optional-arguments)
+
+| Name                 | Description                                           |
+|----------------------|-------------------------------------------------------|
+| `host`               | Host name for the remote data node.                   |
+| `port`               | Port to use on the remote data node.                  |
+| `database`           | Database name where the remote hypertable is created. |
+| `local_user`         | User name on the local node.                          |
+| `remote_user`        | User name on the remote data node.                    |
+| `password`           | Password on the remote data node.                     |
+| `if_not_exists`      | Do not fail if the hyper table already exists         |
+| `bootstrap_database` | Database to bootstrap.                                |
+| `bootstrap_user`     | User to used for bootstrapping.                       |
+| `bootstrap_password` | Password for user use when bootstrapping.             |
+
+##### Parameter Description
+
+* `host` - the name of the host to use for the remote data node. This
+  defaults to `'localhost'`.
+
+* `post` - Port to use for the remote data node. This defaults to the
```

Comment:
@mfreed Good point. I feel the call with defaults should always fail. The default host is the local host and the default database is the same database name as current, thus it will try to add the same node/database to itself, which will fail if `if_not_exists` is not set.

Reply:
I guess I should have been clearer:  From this discussion, my reaction is that `host` should be *required* not optional.

Reply:
I think it will be confusing, yes. Since data nodes are really only sensible if elsewhere, we can have the host and port as required arguments. If a user really want to have things locally, she can always provide the local address herself.

Reply:
I think with port as optional makes sense given default to postgres, was more commenting about host field.

Reply:
Right.

Reply:
Will handle this as a separate PR.